### PR TITLE
Refactor and test get_first_observation_x

### PR DESCRIPTION
**Pull request by hnformentin**
**Issue**
Resolves partially #317

**Approach**
- Remove small chuncks of code out of the main callbacks and test them

### CODE REVIEW

File: tests/views/test_response_correlation.py [hunk 1]
```diff
@@ -131,3 +135,26 @@ def test_show_respo_with_obs(mock_data, dash_duo):
...
+)
+def test_get_first_observation_x_valid(df_observation, expected):
+    result = _get_first_observation_x(df_observation)
+    assert result == expected
+
+
+def test_get_first_observation_x_invalid():
+    df_observation = pd.DataFrame([int(1)], columns=["x_axis"])
+    with pytest.raises(ValueError, match="invalid obs_data type"):
+        _get_first_observation_x(df_observation)
```
Comment (hnformentin) [inline tests/views/test_response_correlation.py:160]: I tried to have conditional raises with pytest parametrize, i.e. having `(pd.DataFrame([int(1)], columns=["x_axis"]), ValueError)`, but it failed with `Failed: [undefined]ValueError:`

**Comment by hnformentin** [comment]
I think is better to merge small changes at a time because it makes easier reviewing and addressing feedbacks.
So I suggest to have sequential PRs until closing the issue #317 instead of trying to do it in only one.

**Review by frode-aarstad — APPROVED** [review]
Looks good 👍🏻

**Comment by hnformentin** [comment]
Thanks!

**Pull request merged**